src/api/query.rs

```diff
@@ -96,117 +102,146 @@ impl QueryHelper {
         };
         Ok((block_info, amount))
     }
 
+    pub async fn query_consensus_info(&self) -> ApiResult<ConsensusInfo> {
+        map_query_result(
+            self.client.clone().get_consensus_info().await,
+            ApiError::UnexpectedSdkError,
+        )
+    }
+
     pub async fn query_account_info_by_address(
         &self,
         addr: AccountAddress,
-        block_hash: &BlockHash,
+        block_id: impl IntoBlockIdentifier,
     ) -> ApiResult<AccountInfo> {
+        let acc_id = v2::AccountIdentifier::Address(addr);
         map_query_result(
-            self.client.clone().get_account_info(addr, block_hash).await,
+            self.client.clone().get_account_info(&acc_id, block_id).await.map(|x| x.response),
             ApiError::NoAccountsMatched,
         )
     }
 
-    pub async fn query_block_info_by_hash(&self, block_hash: &BlockHash) -> ApiResult<BlockInfo> {
+    pub async fn query_block_info_by_hash(
+        &self,
+        block_id: impl IntoBlockIdentifier,
+    ) -> ApiResult<BlockInfo> {
         map_query_result(
-            self.client.clone().get_block_info(block_hash).await,
+            self.client.clone().get_block_info(block_id).await.map(|x| x.response),
             ApiError::NoBlocksMatched,
         )
     }
 
-    pub async fn query_block_summary_by_hash(
+    pub async fn query_block_item_summary(
         &self,
-        block_hash: &BlockHash,
-    ) -> ApiResult<BlockSummary> {
-        map_query_result(
-            self.client.clone().get_block_summary(block_hash).await,
-            ApiError::NoBlocksMatched,
-        )
+        block_id: impl IntoBlockIdentifier,
+    ) -> ApiResult<Vec<BlockItemSummary>> {
+        let event_stream =
+            self.client.clone().get_block_transaction_events(block_id).await.unwrap().response;
+        let events: Result<Vec<BlockItemSummary>, Status> = event_stream.try_collect().await;
+        Ok(events.map_err(RPCError::CallError)?)
+    }
+
+    pub async fn query_block_special_events(
+        &self,
+        block_id: impl IntoBlockIdentifier,
+    ) -> ApiResult<Vec<SpecialTransactionOutcome>> {
+        let event_stream =
+            self.client.clone().get_block_special_events(block_id).await.unwrap().response;
+        let events: Result<Vec<SpecialTransactionOutcome>, Status> =
+            event_stream.try_collect().await;
+        Ok(events.map_err(RPCError::CallError)?)
     }
 
-    pub async fn query_reward_status_by_hash(
+    pub async fn query_tokenomics_info(
         &self,
-        block_hash: &BlockHash,
+        block_id: impl IntoBlockIdentifier,
     ) -> ApiResult<RewardsOverview> {
         map_query_result(
-            self.client.clone().get_reward_status(block_hash).await,
+            self.client.clone().get_tokenomics_info(block_id).await.map(|x| x.response),
             ApiError::NoBlocksMatched,
         )
     }
 
+    pub async fn query_block_hash_from_height(&self, height: i64) -> ApiResult<BlockHash> {
```

Comment:
Documentation please.

Also we have types `AbsoluteBlockHeight` that should be used instead of i64.

Reply:
The input is an `i64` since it comes from the Rosetta API struct `PartialBlockIdentifier` that contains an optional `i64` index field.